test/node.go

```diff
@@ -440,6 +500,47 @@ func NewNetwork(accounts *env.AccountsConfig, gc *genesis.Config, ec *eth.Config
 	}
 
 	shutdown := func() {
+		log.Info("Shutting down network from test")
+		for _, err := range network.Shutdown() {
+			fmt.Println(err.Error())
+		}
+	}
+
+	return network, shutdown, nil
+}
+
+// AddNonValidatorNodes Adds non-validator nodes to the network with the specified sync mode.
+func AddNonValidatorNodes(network Network, ec *eth.Config, numNodes uint64, syncmode downloader.SyncMode) (Network, func(), error) {
```

Comment:
Loving the approach to adding non validator nodes, very neat (apart from the internal duplication but I'm sure that could be easily solved later if we want to) 👌